lib/fog/openstack/requests/compute/delete_flavor_metadata.rb

```diff
@@ -0,0 +1,30 @@
+module Fog
+  module Compute
+    class OpenStack
+      class Real
+        def delete_flavor_metadata(flavor_ref, key)
+          request(
+            :expects => 200,
+            :method  => 'DELETE',
+            :path    => "flavors/#{flavor_ref}/os-extra_specs/#{key}"
+          )
+        end
+      end
+
+      class Mock
+        def delete_flavor_metadata(_flavor_ref, _key)
+          response = Excon::Response.new
+          response.status = 200
+          response.headers = {
+            "X-Compute-Request-Id" => "req-fdc6f99e-55a2-4ab1-8904-0892753828cf",
+            "Content-Type"         => "application/json",
+            "Content-Length"       => "356",
+            "Date"                 => Date.new
+          }
+          response.body = nil
+          response
+        end
+      end # mock
```

Comment:
I understand there are some legacy files using the `end # blah` but I believe this is not a good practice.
My stand is that any good IDE/text editor can collapse blocks easily if needed but more importantly this create a coupling between the comment and the code which in turn makes any changes difficult. Same for all three below.